frontend/controller/backend.js

```diff
@@ -23,70 +23,34 @@ function signJSON (json, keypair) {
-          if (!msg.data) throw new Error('malformed message: ' + JSON.stringify(msg))
-          switch (msg.type) {
-            case RESPONSE_TYPE.ENTRY:
-              // We MUST use 'state/enqueueHandleEvent' here to ensure handleEvent()
-              // is called AFTER any currently-running calls to syncContractWithServer().
-              // Calling via SBP also makes it simple to implement 'test/backend.js'
-              sbp('state/enqueueHandleEvent', GIMessage.deserialize(msg.data))
-              break
-            case RESPONSE_TYPE.SUB:
-            case RESPONSE_TYPE.UNSUB:
-            case RESPONSE_TYPE.PUB: // .PUB can be used to send ephemeral messages outside of any contract logs
-              console.debug(`NOTE: ignoring websocket event ${msg.type} in room:`, msg.data)
-              break
-            default:
-              console.error('SOCKET UNHANDLED EVENT!', msg) // TODO: this
-          }
-        }
+export function createGIPubSubClient (url: string, options: Object): Object {
+  return createClient(url, {
+    ...options,
+    messageHandlers: {
+      [NOTIFICATION_TYPE.ENTRY] (msg) {
+        // We MUST use 'state/enqueueHandleEvent' here to ensure handleEvent()
+        // is called AFTER any currently-running calls to syncContractWithServer().
+        // Calling via SBP also makes it simple to implement 'test/backend.js'
+        sbp('state/enqueueHandleEvent', GIMessage.deserialize(msg.data))
       }
-      // TODO: handle going offline event
-    })
-    serverSocket.on('reconnected', () => {
-      const contractIDs = Object.keys(contractSubscriptions)
-      console.log('websocket connection re-established. re-joining:', contractIDs)
-      contractIDs.forEach(id => serverSocket.sub(id))
-    })
+    }
   })
 }
 
-// Keep pubsub in sync (logged into the right "rooms") with store.state.contracts
+// Keep pubsub in sync (logged into the right "rooms") with 'store.state.contracts'.
 sbp('okTurtles.events/on', CONTRACTS_MODIFIED, async (contracts) => {
-  const subscribedIDs = Object.keys(contractSubscriptions)
+  const client = sbp('okTurtles.data/get', PUBSUB_INSTANCE)
+  const subscribedIDs = [...client.subscriptionSet]
   const currentIDs = Object.keys(contracts)
   const leaveSubscribed = intersection(subscribedIDs, currentIDs)
   const toUnsubscribe = difference(subscribedIDs, leaveSubscribed)
   const toSubscribe = difference(currentIDs, leaveSubscribed)
   try {
     for (const contractID of toUnsubscribe) {
-      const res = await serverSocket.unsub(contractID)
-      delete contractSubscriptions[contractID]
-      console.debug(`[Backend] unsubscribed ${contractID}`, res)
+      client.unsub(contractID)
     }
     for (const contractID of toSubscribe) {
-      const res = await serverSocket.sub(contractID)
-      contractSubscriptions[contractID] = true
-      console.debug(`[Backend] subscribed ${contractID}`, res)
+      client.sub(contractID)
```

Comment:
I'm a bit surprised... are neither Flow nor Standard giving you an error here?

It is my understanding that ever function labeled `async` must have at least one corresponding `await` within it. And yet I see no `await` any longer inside of this function? So if `sub` and `unsub` do not return promises, and therefore no `await` needs to be called, then `async` should be removed from the function identifier on line 41.

Reply:
No error for now, likely because our Flow and Standard setups are configured to be quite lenient.

While every function using `await` must be labeled `async`, the opposite is not strictly required, although it's arguably less confusing.

According to the [require-await](https://eslint.org/docs/rules/require-await) rule (which could be enabled to report such cases):
> Asynchronous functions that don't use await might not need to be asynchronous functions and could be the unintentional result of refactoring.

So it's probably best to avoid, however removing the `async` keyword will change the return type of the function.

Reply:
@snowteamer Feel free to both enable the `require-await` rule, and also remove the `async` keyword, because the return type here doesn't matter. This is just an event handler for an event. The return value is ignored (you can confirm that per `okTurtles.events` source).

Reply:
Enabling this rule globally results in four other similar errors being found, so maybe it would be better to just remove the `async` keyword here for now?

Reply:
Can you check if fixing those other errors breaks anything? If not, feel free to remove `async` in those other locations too and enable the rule. It seems (to me) like a useful rule that used to be enabled, and maybe got disabled by a change in Standard.

Reply:
Sure. In this PR or in a quick new one?

Reply:
I guess we can do it in a separate PR. Just opened #1058 for it.